src/submodule.c

```diff
@@ -1941,7 +1943,17 @@ static int submodule_read_config(git_submodule *sm, git_config *cfg)
 		goto cleanup;
 	}
 
-	if ((error = get_value(&value, cfg, &key, sm->name, "ignore")) == 0) {
+	/* For `ignore', first check the config of the parent repository. */
+	if (git_repository_config_snapshot(&repo_snap_cfg, sm->repo) == 0) {
+		repo_cfg_cleanup = true;
```

Comment:
No need to set a boolean here - instead, we can just set `repo_snap_cfg` to `NULL` in its initialization and then `free` it without a test.

Reply:
Makes sense. Will initialize `repo_snap_cfg` to `NULL`. Thanks!